keyboards/chidori/keymaps/oled_sample/keymap.c

```diff
@@ -0,0 +1,239 @@
+      _______, _______, _______, _______, _______, _______,           _______, _______, _______, _______, _______, _______,
+      _______, _______, _______, _______, _______, _______,           _______, _______, KC_HOME, KC_PGDN, KC_PGUP, KC_END
+    )
+};
+// clang-format on
+
+layer_state_t layer_state_set_user(layer_state_t state) {
+    return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        case QWERTY:
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(_QWERTY);
+            }
+            return false;
+        case COLEMAK:
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(_COLEMAK);
+            }
+            return false;
+        case DVORAK:
+            if (record->event.pressed) {
+                set_single_persistent_default_layer(_DVORAK);
+            }
+            return false;
+    }
+    return true;
+}
+
+bool led_update_user(led_t led_state) {
+    board_set_led_by_index(0, LED_YELLOW, led_state.caps_lock);
+    board_set_led_by_index(1, LED_YELLOW, led_state.scroll_lock);
+
+    return false;
+}
+
+#ifdef OLED_DRIVER_ENABLE
+
+const char *read_layer_state(void) {
+    // Layer names
+    static const char *LAYER_NAME_QWERTY  = "Qwerty";
+    static const char *LAYER_NAME_COLEMAK = "Colemak";
+    static const char *LAYER_NAME_DVORAK  = "Dvorak";
+    static const char *LAYER_NAME_LOWER   = "Lower";
+    static const char *LAYER_NAME_RAISE   = "Raise";
+    static const char *LAYER_NAME_ADJUST  = "Adjust";
+    static const char *LAYER_NAME_UNDEF   = "Undef";
+    // Default layer
+    const char *default_layer = LAYER_NAME_UNDEF;
+    switch (get_highest_layer(default_layer_state)) {
+        case _QWERTY:
+            default_layer = LAYER_NAME_QWERTY;
+            break;
+        case _COLEMAK:
+            default_layer = LAYER_NAME_COLEMAK;
+            break;
+        case _DVORAK:
+            default_layer = LAYER_NAME_DVORAK;
+            break;
+        default:
+            break;
+    }
+    // Current layer
+    const char *layer = LAYER_NAME_UNDEF;
+    switch (get_highest_layer(layer_state)) {
+        case 0:  // Default layer
+            layer = default_layer;
+            break;
+        case _LOWER:
+            layer = LAYER_NAME_LOWER;
+            break;
+        case _RAISE:
+            layer = LAYER_NAME_RAISE;
+            break;
+        case _ADJUST:
+            layer = LAYER_NAME_ADJUST;
+            break;
+        default:
+            layer = LAYER_NAME_UNDEF;
+            break;
+    }
+    static char layer_str[24];
+    snprintf(layer_str, sizeof(layer_str), "Layer: %s", layer);
+    return layer_str;
+}
+
+void oled_task_user(void) {
+    // If you want to change the display of OLED, you need to change here
+    oled_write_ln(read_layer_state(), false);
+}
+#endif
```

Comment:
Well, glad to see the newer function used, you're still using the older code for it.  While that may be fine.... you're missing out on the PROGMEM functionality, which will reduce memory usage, since you're not modifying a string in memory. 

It may be better to do something like this: 

```suggestion
void oled_task_user(void) {
    oled_write_P("Layer: "), false);
    switch (get_highest_layer(layer_state)) {
        case 0:  // Default layer
            switch (get_highest_layer(default_layer_state)) {
                case _QWERTY:
                    oled_write_ln_P("Qwerty", false);
                    break;
                case _COLEMAK:
                    oled_write_ln_P("Colemak", false);
                    break;
                case _DVORAK:
                    oled_write_ln_P("Dvorak", false);
                    break;
            }
            break;
        case _LOWER:
            oled_write_ln_P("Lower", false);
            break;
        case _RAISE:
            oled_write_ln_P("Raise", false);
            break;
        case _ADJUST:
            oled_write_ln_P("Adjust", false);
            break;
        default:
            oled_write_ln_P("Undef", false);
            break;
    }
}
```

Reply:
Thanks for your review! I didn't know about PROGMEM, so it was very helpful.

In addition, I split the process of displaying the layer state to a function so that users can easily switch what shows on OLED.

Reply:
You're very welcome!  And yeah, I've ... ran into issues with not using it with the oled stuff myself.